src/components/Map.js

```diff
@@ -62,6 +65,19 @@ export class UnconnectedMap extends Component {
 					this.map.removeLayer(this.leafletPolylineLayers[layer.ukey]);
 				}
 				this.updatePolylineLayer(layer.data, layer.color, layer.ukey);
+			} else if (layer.rendering === RENDERING_RELATIONS) {
+				ukeyRelationshipArray.push(layer.ukey);
+				if (this.leafletRelationshipLayers[layer.ukey]) {
+					// todo find a way of updating the polyline layer instead of delete & recreate
+					this.map.removeLayer(this.leafletRelationshipLayers[layer.ukey]);
+				}
+				this.updateRelationsLayer(layer.data, layer.relationshipData, layer.relationshipColor, layer.ukey);
+				// have to use separate marker layer
+				ukeyMarkerArray.push(layer.ukey);
```

Comment:
that's not great cos i copy-pasted this part
i'm not even sure that this is the right thing to do but if it is, from code perspective i'd like move this to a small function